core/common/src/SinkExt.kt

```diff
@@ -0,0 +1,272 @@
+        else if (v < 1000000000000L)
+            if (v < 10000000000L)
+                if (v < 1000000000L) 9
+                else 10
+            else if (v < 100000000000L) 11
+            else 12
+        else if (v < 1000000000000000L)
+            if (v < 10000000000000L) 13
+            else if (v < 100000000000000L) 14
+            else 15
+        else if (v < 100000000000000000L)
+            if (v < 10000000000000000L) 16
+            else 17
+        else if (v < 1000000000000000000L) 18
+        else 19
+    if (negative) {
+        ++width
+    }
+
+    writeToInternalBuffer { buffer ->
+        val tail = buffer.writableSegment(width)
+        val data = tail.data
+        var pos = tail.limit + width // We write backwards from right to left.
+        while (v != 0L) {
+            val digit = (v % 10).toInt()
+            data[--pos] = HEX_DIGIT_BYTES[digit]
+            v /= 10
+        }
+        if (negative) {
+            data[--pos] = '-'.code.toByte()
+        }
+
+        tail.limit += width
+        buffer.size += width.toLong()
+    }
+}
+
+/**
+ * Writes [long] to this sink in hexadecimal form (i.e., as a string in base 16).
+ *
+ * Resulting string will not contain leading zeros, except the `0` value itself.
+ *
+ * @param long the long to be written.
+ *
+ * @throws IllegalStateException when the sink is closed.
+ */
+@OptIn(DelicateIoApi::class)
+public fun Sink.writeHexadecimalUnsignedLong(long: Long) {
+    var v = long
+    if (v == 0L) {
+        // Both a shortcut and required since the following code can't handle zero.
+        writeByte('0'.code.toByte())
+        return
+    }
+
+    // Mask every bit below the most significant bit to a 1
+    // http://aggregate.org/MAGIC/#Most%20Significant%201%20Bit
+    var x = v
+    x = x or (x ushr 1)
+    x = x or (x ushr 2)
+    x = x or (x ushr 4)
+    x = x or (x ushr 8)
+    x = x or (x ushr 16)
+    x = x or (x ushr 32)
+
+    // Count the number of 1s
+    // http://aggregate.org/MAGIC/#Population%20Count%20(Ones%20Count)
+    x -= x ushr 1 and 0x5555555555555555
+    x = (x ushr 2 and 0x3333333333333333) + (x and 0x3333333333333333)
+    x = (x ushr 4) + x and 0x0f0f0f0f0f0f0f0f
+    x += x ushr 8
+    x += x ushr 16
+    x = (x and 0x3f) + ((x ushr 32) and 0x3f)
+
+    // Round up to the nearest full byte
+    val width = ((x + 3) / 4).toInt()
+
+    writeToInternalBuffer { buffer ->
+        val tail = buffer.writableSegment(width)
+        val data = tail.data
+        var pos = tail.limit + width - 1
+        val start = tail.limit
+        while (pos >= start) {
+            data[pos] = HEX_DIGIT_BYTES[(v and 0xF).toInt()]
+            v = v ushr 4
+            pos--
+        }
+        tail.limit += width
+        buffer.size += width.toLong()
+    }
+}
+
+/**
+ * Writes am unsigned byte to this sink.
+ *
+ * @param byte the byte to be written.
+ *
+ * @throws IllegalStateException when the sink is closed.
+ */
+public fun Sink.writeByte(byte: UByte) {
```

Comment:
just want to understand: why it's not `writeUByte`? (same for other unsigned writes)
now it's a little inconsistent with `read` part: `readUByte(): UByte` but `writeByte(value: UByte)`

Reply:
I don't have a strong opinion here. It should probably follow the common naming convention and be `writeU<type>`, as you suggested.